tests/integration/schema.test.js

```diff
@@ -0,0 +1,141 @@
+      address: FullNodeEndpoint,
+    });
+    account = dock.keyring.addFromUri(TestAccountURI);
+    dock.setAccount(account);
+    pair = dock.keyring.addFromUri(firstKeySeed);
+    publicKey = getPublicKeyFromKeyringPair(pair);
+    dockDID = createNewDockDID();
+    keyDetail = createKeyDetail(publicKey, dockDID);
+    await dock.sendTransaction(dock.did.new(dockDID, keyDetail));
+    blobId = randomAsHex(DockBlobIdByteSize);
+
+    vcInvalid.credentialSchema = {
+      id: blobId,
+      type: 'JsonSchemaValidator2018',
+    };
+
+    // Write invalid format blob
+    invalidFormatBlobId = randomAsHex(DockBlobIdByteSize);
+    await dock.sendTransaction(dock.blob.new({
+      id: invalidFormatBlobId,
+      blob: stringToHex('hello world'),
+      author: getHexIdentifierFromDID(dockDID),
+    }, pair), false);
+
+    // Write schema blob
+    const blobStr = JSON.stringify(exampleSchema);
+    await dock.sendTransaction(dock.blob.new({
+      id: blobId,
+      blob: stringToHex(blobStr),
+      author: getHexIdentifierFromDID(dockDID),
+    }, pair), false);
+
+    done();
+  }, 120000);
+
+  afterAll(async () => {
+    await dock.disconnect();
+  }, 120000);
+
+  test('setSignature will only accept signature of the supported types and set the signature key of the object.', async () => {
+    const schema = new Schema();
+    schema.setAuthor(dockDID);
+    schema.name = 'AlumniCredSchema';
+    await schema.setJSONSchema(exampleSchema);
+    const msg = dock.blob.getSerializedBlob(schema.toBlob());
+    const pk = getPublicKeyFromKeyringPair(pair);
+    const sig = new SignatureSr25519(msg, pair);
+    schema.setSignature(sig);
+    expect(schema.signature).toBe(sig);
+  });
+
+  test('sign will generate a signature on the schema detail, this signature is verifiable.', async () => {
+    const schema = new Schema();
+    schema.setAuthor(dockDID);
+    await schema.setJSONSchema(exampleSchema);
+    schema.name = 'AlumniCredSchema';
+    schema.sign(pair, dock.blob);
+    expect(!!schema.signature).toBe(true);
+  });
+
+  test('getSchema will return schema in correct format.', async () => {
+    await expect(Schema.getSchema(blobId, dock)).resolves.toMatchObject({
+      ...exampleSchema,
+      id: blobId,
+      author: hexDIDToQualified(getHexIdentifierFromDID(dockDID)),
+    });
+  }, 120000);
+
+  test('getSchema throws error when schema not in correct format.', async () => {
+    await expect(Schema.getSchema(invalidFormatBlobId, dock)).rejects.toThrow(/Incorrect schema format/);
+  }, 120000);
+
+  test('getSchema throws error when no blob exists at the given id.', async () => {
+    await expect(Schema.getSchema(createNewDockBlobId(), dock)).rejects.toThrow(/does not exist/);
+  }, 120000);
+
+  test('Utility method verifyCredential should check if schema is incompatible with the credentialSubject.', async () => {
+    await expect(
+      verifyCredential(
+        vcInvalid, null, false, false, undefined, { notDock: dock },
```

Comment:
The 3rd arg to `verifyCredential` should be kept true as we are compacting JSON-LD